Tests/UnitTests/Networking/Backend/__Snapshots__/BackendPostAdServicesTokenTests/iOS14-testPostAdServicesCallsHttpClient.6.json

```diff
@@ -0,0 +1,12 @@
+{
```

Comment:
Hmm something's weird here. The fact that there are 6 files means the operation is making 6 network requests.

Reply:
that's....not desirable... 

will take a look back here on monday. thanks for the info!

Reply:
it might not be your fault though. I saw the same issue in #1561, but I haven't been able to reproduce locally.
Since you did reproduce it it would be nice if you could debug why the request is happening multiple times 🙏🏻 or maybe it's something else entirely, but hopefully we can figure it out. I have 2 flights on Monday but might have some time to help out with this.

Reply:
@NachoSoto i confirmed the operation is not requesting multiple times, both when running this unit test and in the sample app... any other thoughts on why that could be happening?

Reply:
I couldn't reproduce locally though.

> Perhaps while taking snapshots, the app is becoming active multiple times quickly,

Can you explain what you mean by this? The duplicated snapshots come from the same test.

Reply:
I have almost no familiarity with how our snapshots are generated, I haven't had much time to play around with them. But my thinking was that if `appDidBecomeActive` gets called multiple times during the same test run, it would trigger multiple calls to this method. Same if we called configure multiple times in the same test run. 
Either of those would potentially start multiple calls to the method that fires the backend request

Reply:
Snapshots are generated every time `HTTPClient.perform(request)` is called (except in these tests, it's `MockHTTPClient`). So the presence of multiple files signals that the method is called multiple times throughout the lifetime of the test.

Reply:
I wonder if this was just a bad test run with a previous change. @beylmk I recommend removing the entire `__Snapshots__` directory and running the tests again. I'm doing that locally (even on iOS 12) and I don't reproduce the issue, it's only making the appropriate number of requests now.

Reply:
thanks @aboedo and @NachoSoto for the thoughts... i'll try deleting the directory and rerunning!